journal/src/test/java/io/zeebe/journal/file/JournalTest.java

```diff
@@ -497,6 +508,77 @@ public void shouldInvalidateAllEntries() throws Exception {
     assertThat(reader.hasNext()).isFalse();
   }
 
+  @Test
+  public void shouldIgnorePartiallyWrittenRecordOnStartup() throws Exception {
+    // given
+    data.wrap("000".getBytes(StandardCharsets.UTF_8));
+    final var firstRecord = journal.append(data);
+    final var secondRecord = journal.append(data);
+    assertThat(directory.toFile().listFiles()).hasSize(1);
+    assertThat(directory.toFile().listFiles()[0].listFiles()).hasSize(1);
+    final File log = directory.toFile().listFiles()[0].listFiles()[0];
+
+    final File tempFile = new File(temp.toFile(), "rand");
+    assertThat(tempFile.createNewFile()).isTrue();
+    assertThat(corruptFile(tempFile, log, secondRecord.index())).isTrue();
```

Comment:
Nit: May be move all these to `corruptFile()`.

Reply:
If we keep this function as static we can reuse this logic in other places. The other PR has tests that require similar logic to corrupt records